src/auth/ldap_krb/KrbServiceHandler.cpp

```diff
@@ -0,0 +1,253 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 Daniel Oliveira <doliveira@suse.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+/* Include order and names:
+ * a) Immediate related header
+ * b) C libraries (if any),
+ * c) C++ libraries,
+ * d) Other support libraries
+ * e) Other project's support libraries
+ *
+ * Within each section the includes should
+ * be ordered alphabetically.
+ */
+
+#include "KrbServiceHandler.hpp"
+
+#include <errno.h>
+#include <sstream>
+
+#include "KrbProtocol.hpp"
+#include "common/config.h"
+#include "common/debug.h"
+
+
+#define dout_subsys ceph_subsys_auth
+#undef dout_prefix
+#define dout_prefix *_dout << "krb5/gssapi service: " << entity_name <<  " : "
+
+
+int KrbServiceHandler::handle_request(bufferlist::const_iterator& indata, 
+                                      bufferlist& buff_list, 
+                                      uint64_t& global_id, 
+                                      AuthCapsInfo& caps, 
+                                      uint64_t* auid) 
+{
+  constexpr auto SUBSYSTEM_ID(20);
+  constexpr auto SUBSYSTEM_ZERO(0);
```

Comment:
this is not defining subsystem's ids but log levels, which could be specified in ldout directly.